modules/web3/helpers/getDefaultProvider.ts

```diff
@@ -13,16 +13,19 @@ import { ethers, providers } from 'ethers';
 ethers.utils.Logger.setLogLevel(ethers.utils.Logger.levels.ERROR);
 
 export const getDefaultProvider = (
-  network: SupportedNetworks | string | undefined,
-  optionsOverrides?: Record<string, string>
-): providers.BaseProvider => {
-  const options = {
-    infura: config.INFURA_KEY,
-    alchemy: config.ALCHEMY_KEY,
-    pocket: config.POCKET_KEY,
-    etherscan: config.ETHERSCAN_KEY,
-    ...(optionsOverrides || {})
+  network: SupportedNetworks | string | undefined
+): providers.FallbackProvider => {
+    const infuraProvider = new ethers.providers.InfuraProvider(network, config.INFURA_KEY);
+    const alchemyProvider = new ethers.providers.AlchemyProvider(network, config.ALCHEMY_KEY);
+    const pocketProvider = new ethers.providers.PocketProvider(network, config.POCKET_KEY);
+    const etherscanProvider = new ethers.providers.EtherscanProvider(network, config.ETHERSCAN_KEY);
+  
+    const provider = new ethers.providers.FallbackProvider([
+      { provider: infuraProvider, priority: 2 },
+      { provider: alchemyProvider, priority: 1 },
+      { provider: pocketProvider, priority: 3 },
+      { provider: etherscanProvider, priority: 3 }
+    ], 1); //Quorum of 1
```

Comment:
cool, so this will always try alchemy first, and if it is successful then we're good, right?